osprey/processes/wps_parameters.py

```diff
@@ -0,0 +1,130 @@
+# Processor imports
+from pywps import (
+    Process,
+    LiteralInput,
+)
+from pywps.app.Common import Metadata
+
+# Tool imports
+from rvic.parameters import parameters
+from rvic.core.config import read_config
+from wps_tools.utils import (
+    collect_output_files,
+    log_handler,
+)
+from wps_tools.io import (
+    log_level,
+    nc_output,
+)
+
+# Library imports
+import logging
+import os
+
+pywps_logger = logging.getLogger("PYWPS")
+
+
+class Parameters(Process):
+    def __init__(self):
+        self.status_percentage_steps = {
+            "start": 0,
+            "dry_run": 5,
+            "process": 10,
+            "collect_files": 90,
+            "build_output": 95,
+            "complete": 100,
+        }
+        inputs = [
+            LiteralInput(
+                "config",
+                "Configuration",
+                abstract="Path to input configuration file or input dictionary",
+                data_type="string",
+            ),
+            LiteralInput(
+                "np",
+                "numofproc",
+                default=1,
+                abstract="Number of processors used to run job",
+                data_type="integer",
+            ),
+            LiteralInput(
+                "version",
+                "Version",
+                default=True,
+                abstract="Return RVIC version string",
+                data_type="boolean",
+            ),
+        ]
+        outputs = [
+            nc_output,
+        ]
+
+        super(Parameters, self).__init__(
+            self._handler,
+            identifier="parameters",
+            title="Parameters",
+            abstract="Develop impulse response functions using inputs from a "
+            "configuration file or dictionary",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def collect_args(self, request):
+        config = request.inputs["config"][0].data
+        np = request.inputs["np"][0].data
+        return (config_file, np)
+
+    def get_outfile(self, config):
+        if isinstance(config, dict):
+            config_dict = config
+        else:
+            config_dict = read_config(config)
+        outdir = os.path.join(config_dict["OPTIONS"]["CASE_DIR"], "params")
+        (param_file,) = collect_output_files("rvic", outdir)
+        return os.path.join(outdir, param_file)
+
+    def _handler(self, request, response):
+        if request.inputs["version"][0].data:
+            from rvic import version
```

Comment:
Why are we running this import in this manner?

Reply:
This is the way it's [done in RVIC](https://github.com/UW-Hydro/RVIC/blob/5987da6d33cea8063c848b60215353915ac2bb10/scripts/rvic#L31). I could move the import to the top if that'd be better.

Reply:
This is the way it's [done in RVIC](https://github.com/UW-Hydro/RVIC/blob/5987da6d33cea8063c848b60215353915ac2bb10/scripts/rvic#L31). I could move the import to the top.

Reply:
Hmm they _do_ have an arg for it... Ok I think you were right to put it in in this manner.

Reply:
The import statement in `wps_convolution` is placed at the top. Should we be matching the place it is imported?

Reply:
Good point @sum1lim, it would be nice to have the files matching. I believe having the imports at the top is standard so we can move them there.